server/src/main/java/com/unosquare/carmigo/service/JourneyService.java

```diff
@@ -1,29 +1,85 @@
+import com.unosquare.carmigo.repository.JourneyRepository;
+import com.unosquare.carmigo.util.MapperUtils;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Service;
 
-import java.util.Map;
+import javax.persistence.EntityManager;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
 
 @Service
-public class JourneyService {
+@RequiredArgsConstructor
+public class JourneyService
+{
+    private final JourneyRepository journeyRepository;
+    private final ModelMapper modelMapper;
+    private final ObjectMapper objectMapper;
+    private final EntityManager entityManager;
 
-    public static String getJourneyParameters(final Map<String, String> paramMap)
+    public GrabJourneyDTO getJourneyById(final int id)
     {
-        if (!paramMap.isEmpty()) {
-            final Map.Entry<String, String> entry = paramMap.entrySet().iterator().next();
-            final String key = entry.getKey();
-            final String value = entry.getValue();
-
-            switch (key) {
-                case "passenger_id":
-                    return "Here are all journeys for passenger " + value;
-                case "driver_id":
-                    return "Here are all journeys for driver " + value;
-                default:
-                    return "Error retrieving journeys";
+        try {
+            final Journey journey = journeyRepository.findJourneyById(id);
+            return modelMapper.map(journey, GrabJourneyDTO.class);
+        } catch (final IllegalArgumentException ex) {
+            throw new ResourceNotFoundException("Journey id " + id + " not found.");
+        }
+    }
+
+    public List<GrabJourneyDTO> getJourneys()
+    {
+        final List<Journey> result = journeyRepository.findAll();
+        return MapperUtils.mapList(result, GrabJourneyDTO.class, modelMapper);
+    }
+
+    public GrabJourneyDTO createJourney(final CreateJourneyDTO createJourneyDTO)
+    {
+        final Journey journey = modelMapper.map(createJourneyDTO, Journey.class);
+        journey.setCreatedDate(Instant.now());
+        journey.setLocationFrom(entityManager.getReference(Location.class, createJourneyDTO.getLocationIdFrom()));
+        journey.setLocationTo(entityManager.getReference(Location.class, createJourneyDTO.getLocationIdTo()));
+        journey.setDriver(entityManager.getReference(Driver.class, createJourneyDTO.getDriver()));
+        return modelMapper.map(journeyRepository.save(journey), GrabJourneyDTO.class);
+    }
+
+    public GrabJourneyDTO patchJourney(final int journeyId, final int driverId, final JsonPatch patch)
+    {
+        final List<Journey> journeys = journeyRepository.findJourneyByDriverId(driverId);
+        final Optional<Journey> targetJourney = journeys.stream()
+                .filter(journey -> journey.getId() == journeyId)
+                .findFirst();
+        try {
+            if (targetJourney.isPresent()) {
```

Comment:
I would take the check out of the try block and as above:

`if (!targetJourney.isPresent()) { return; }`

You don't need to throw exceptions, just handle an empty result.

exceptions are always a little more epensive in terms of processing.